src/Voron/EnvironmentStateRecord.cs

```diff
@@ -0,0 +1,20 @@
+using System.Collections.Frozen;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Voron.Data.BTrees;
+using Voron.Impl.Journal;
+using Voron.Impl.Paging;
+using Voron.Impl.Scratch;
+
+namespace Voron;
+
+public record EnvironmentStateRecord(
+    Pager.State DataPagerState, 
+    long TransactionId,
+    ImmutableDictionary<long, PageFromScratchBuffer> ScratchPagesTable,
+    long WrittenToJournalNumber,
+    TreeMutableState Root,
+    long NextPageNumber,
+    (JournalFile Current, long Last4KWritePosition) Journal,
+    object ClientState
```

Comment:
What is the expected lifetime of `ClientState`? Is it supposed to be carried between transactions?

Reply:
I see it might be there as long as a storage environment is up

Reply:
Yes, that is the point. It is a way to generate in memory state in a transaction and keep it to the next one